src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp

```diff
@@ -1570,6 +1570,29 @@ class at_icc_start_blood_quickening : public AreaTriggerScript
         }
 };
 
+class at_icc_nerubar_broodkeeper : public AreaTriggerScript
+{
+    public:
+        at_icc_nerubar_broodkeeper() : AreaTriggerScript("at_icc_nerubar_broodkeeper") { }
+
+        bool OnTrigger(Player* player, AreaTriggerEntry const* /*areaTrigger*/)
+        {
+            if (player->IsGameMaster())
+                return false;
+
+            std::list<Creature*> creatureList;
+            player->GetCreatureListWithEntryInGrid(creatureList, NPC_NERUBAR_BROODKEEPER, 100.0f);
+            for (std::list<Creature*>::iterator itr = creatureList.begin(); itr != creatureList.end(); ++itr)
+                if (Creature* creature = *itr)
```

Comment:
this null check might be not needed

Reply:
naah, change everything to foreach
```
for (Creature const& creature : creatureList)
```

Reply:
It's bad?

```
for (std::list<Creature*>::iterator itr = creatureList.begin(); itr != creatureList.end(); ++itr)
                (*itr)->GetMotionMaster()->MoveFall(),
                (*itr)->SetHover(false),
                (*itr)->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, 0),
```

Reply:
i change wait commit.

Reply:
it will not work, missing { } around all instructions.
just make it easy for you, use a foreach, you will have Creature* pointer already
e.g.

```
for (Creature* creature : creatureList)
{
    creature->GetMotionMaster()->MoveFall();
    creature->SetHover(false);
    creature->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, 0);
}
```